### PR TITLE
remove private tests from nightly

### PR DESCRIPTION
**Pull request by guykeller**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37860
fixes: https://github.com/demisto/etc/issues/32512

## Description
Added an option of specifying private tests, that will not be run on nightly.
This possibility will allow us to specify timeout values for private tests.

### CODE REVIEW

**Comment by guykeller** [comment]
Nightly triggered and the HWP scan test indeed did not run:
https://app.circleci.com/pipelines/github/demisto/content/92224/workflows/0e064891-1ab6-4eda-8700-bc351d052918/jobs/380779

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 1]
```diff
@@ -1037,20 +1040,41 @@ def remove_tests_for_non_supported_packs(tests: set, id_set: dict) -> set:
...
+
+def filter_tests(tests: set, id_set: json, is_nightly=False) -> set:
     """
-    Filter tests out from the test set if they are a.Ignored b.Non XSOAR or non-supported packs c. tests of
-    deprecated packs.
+    Filter tests out from the test set if they are:
+    a.Ignored
+    b.Non XSOAR or non-supported packs
+    c. tests of deprecated packs.
+    d. tests of private packs (optional)
```
Comment (yaakovi) [inline Tests/scripts/collect_tests_and_content_packs.py:1061]:
```suggestion
    a. Ignored
    b. Non-XSOAR or non-supported packs
    c. tests of deprecated packs.
    d. tests of private packs (optional)
```

**Review by yaakovi — APPROVED** [review]

**Review by yuvalbenshalom — COMMENTED** [review]
did you se the failures on the regular builds?

**Comment by guykeller** [comment]
@yuvalbenshalom 
I sent you a Slack message earlier today about that.

Since the HWP scan test is running in this build (was removed from skipped so it only runs now, it was only removed from nightly).
It's natural it will fail - HWP isn't in the content repo

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**